public/pages/AnomalyCharts/utils/anomalyChartUtils.ts

```diff
@@ -138,3 +150,281 @@ export const disabledHistoryAnnotations = (
+    //@ts-ignore
+    if (selectedYs.includes(originalYs[i])) {
+      resultYs.push(originalYs[i]);
+      resultZs.push(originalZs[i]);
+      resultTexts.push(originalTexts[i]);
+    }
+  }
+  const updateHeatmapPlotData = {
+    ...cloneDeep(heatmapData),
+    y: resultYs,
+    z: resultZs,
+    text: resultTexts,
+  } as PlotData;
+  return sortHeatmapPlotData(
+    updateHeatmapPlotData,
+    sortType,
+    selectedYs.length
+  );
+};
+
+export const sortHeatmapPlotData = (
+  heatmapData: PlotData,
+  sortType: AnomalyHeatmapSortType,
+  topNum: number
+) => {
+  const originalYs = cloneDeep(heatmapData.y);
+  const originalZs = cloneDeep(heatmapData.z);
+  const originalTexts = cloneDeep(heatmapData.text);
+  const originalValuesToSort =
+    sortType === AnomalyHeatmapSortType.SEVERITY
+      ? cloneDeep(originalZs)
+      : cloneDeep(originalTexts);
+  const funcToAggregate =
+    sortType === AnomalyHeatmapSortType.SEVERITY
+      ? (a: number, b: number) => {
+          return Math.max(a, b);
+        }
+      : (a: number, b: number) => {
+          return a + b;
+        };
+  const yIndicesToSort = [] as object[];
+  for (let i = 0; i < originalYs.length; i++) {
+    yIndicesToSort.push({
+      index: i,
+      //@ts-ignore
+      value: originalValuesToSort[i].reduce(funcToAggregate),
+    });
+  }
+  const sortedYIndices = orderBy(yIndicesToSort, ['value'], 'desc').slice(
+    0,
+    topNum
+  );
+  const resultYs = [] as any[];
+  const resultZs = [] as any[];
+  const resultTexts = [] as any[];
+  for (let i = sortedYIndices.length - 1; i >= 0; i--) {
+    const index = get(sortedYIndices[i], 'index', 0);
+    resultYs.push(originalYs[index]);
+    resultZs.push(originalZs[index]);
+    resultTexts.push(originalTexts[index]);
+  }
+  return {
+    ...cloneDeep(heatmapData),
+    y: resultYs,
+    z: resultZs,
+    text: resultTexts,
+  } as PlotData;
+};
+
+export const updateHeatmapPlotData = (heatmapData: PlotData, update: any) => {
+  return {
+    ...cloneDeep(heatmapData),
+    ...update,
+  } as PlotData;
+};
+
+export const getSelectedHeatmapCellPlotData = (
+  heatmapData: PlotData,
+  selectedX: number,
+  selectedY: number
+) => {
+  const originalZ = cloneDeep(heatmapData.z);
+  const selectedZData = [];
+  //@ts-ignore
+  const selectedValue = originalZ[selectedY][selectedX];
+  for (let i = 0; i < originalZ.length; i++) {
+    const row = [];
+    //@ts-ignore
+    for (let j = 0; j < originalZ[0].length; j++) {
+      if (i === selectedY && j === selectedX) {
+        row.push(selectedValue);
+      } else {
+        row.push(null);
+      }
+    }
+    selectedZData.push(row);
+  }
+  const colorForCell = getColorForValue(selectedValue);
+  //@ts-ignore
+  return [
+    {
+      ...cloneDeep(heatmapData),
+      z: selectedZData,
+      colorscale: [
+        [0, colorForCell],
+        [1, colorForCell],
+      ],
+      opacity: 1,
+      hoverinfo: 'skip',
+      hovertemplate: null,
```

Comment:
why is the `hovertemplate` `null` here? Also, when selecting a cell, it still shows the regular hover template with regular data.

Reply:
if not set to null, it will have extra chevron as there are 2 traces on the heatmap
![extra-chevron](https://user-images.githubusercontent.com/59710443/96323649-e5ed1200-0fd2-11eb-9506-2592a450f060.png)

Reply:
I see yeah, remember this issue. Cool thanks.